grammar.js

```diff
@@ -810,7 +857,7 @@ module.exports = grammar(C, {
 
     // Expressions
 
-    _expression: ($, original) => choice(
+    _expression_not_binary: ($, original) => choice(
```

Comment:
What is the distinction between this one and a normal `_expression`? Bifurcating it like this balloons the parser size and can make it slower which is already a huge issue here. You needed to add a lot of conflicts which is typically a sign that things need to be consolidated or rewritten

Reply:
was an issue in C, top level binary expressions were conflicting with certain function declarations, I ported the changes here